### ISSUE TITLE
Stripped protocol inside url_to_fs

### ISSUE THREAD

**Issue by Veganveins** [intake/filesystem_spec#521]
I see that the `url_to_fs` function inside fsspec.core strips the protocol inside the `else` portion of the if condition, see [here](https://github.com/intake/filesystem_spec/blob/master/fsspec/core.py#L377).

Would we expect that same `fs._strip_protocol(url)` line inside the `if` block of logic as well?

**Comment by martindurant** [comment]
I'm not sure! The main point of the function is to produce the filesystem, as the name suggests - and the resultant filesystem ought to be able to deal with URLs with or without the protocol prefix(es). Some protocol stripping is indeed done in the `True` branch, so best would be to set up a test function with input and expected outputs, and see if something needs to be changed.

**Comment by Veganveins** [comment]
@martindurant thanks for following up on this. In trying to set up a test function with input and expected outputs, I realized that the problem I was having was entirely on my end. (I was accidentally doing something like `path = f's3://{bucket}'`, where `bucket = 's3://bucket-name'`, meaning my path variable had the `s3://` prefix in there twice.

naturally strip_protocol(path) was helping, since I had a double-prefix; but when I passed in a bucket without the s3 prefix I no longer had any issues. so I don't think anything needs to be changed.

I'm fairly new to this open source world, is it customary for me to close this issue, since I resolved it on my end?
